src/components/DatePickerDocked/MonthPicker.tsx

```diff
@@ -0,0 +1,163 @@
+                selectingMonth ? styles.opacity1 : styles.opacity0,
+            ],
+            monthStyle: rest,
+        };
+    }, [selectingMonth, monthPickerStyles]);
+
+    const handleOnChange = useCallback(
+        (month: number) => {
+            onChange(month, 'month');
+        },
+        [onChange],
+    );
+
+    const renderItem = useCallback(
+        ({ item }: { item: number }) => {
+            return (
+                <Month
+                    month={item}
+                    selected={localDate.getMonth() === item}
+                    onPressMonth={handleOnChange}
+                    monthStyles={monthStyle}
+                />
+            );
+        },
+        [localDate, handleOnChange, monthStyle],
+    );
+
+    if (!selectingMonth) {
+        return null;
+    }
+
+    return (
+        <View style={containerStyle} pointerEvents={selectingMonth ? 'auto' : 'none'}>
+            <HorizontalDivider />
+            <FlatList<number>
+                ref={flatList}
+                style={styles.list}
+                data={months}
+                renderItem={renderItem}
+                keyExtractor={item => `${item}`}
+            />
+        </View>
+    );
+}
+
+function MonthPure({
+    month,
+    selected,
+    onPressMonth,
+    monthStyles,
+}: {
+    month: number;
+    selected: boolean;
+    onPressMonth: (newMonth: number) => any;
+    monthStyles: Record<string, any>;
+}) {
+    const { TouchableRipple, Text, View, Icon } = useMolecules();
+    const montLocalStyles = useComponentStyles('DatePickerDocked_Month', monthStyles, {
+        states: {
+            selected,
+        },
+    });
+    const {
+        containerStyle,
+        monthInnerStyle,
+        monthLabelStyle,
+        monthButtonStyle,
+        accessibilityState,
+    } = useMemo(() => {
+        const { month: monthStyle, monthInner, monthLabel, monthButton } = montLocalStyles;
+
+        return {
+            containerStyle: monthStyle,
+            monthInnerStyle: monthInner,
+            monthLabelStyle: monthLabel,
+            monthButtonStyle: monthButton,
+            accessibilityState: { selected },
+        };
+    }, [selected, montLocalStyles]);
+
+    const handleMonthPress = useCallback(() => {
+        onPressMonth(month);
+    }, [onPressMonth, month]);
+
+    return (
+        <View style={containerStyle}>
+            <TouchableRipple
+                onPress={handleMonthPress}
+                accessibilityRole="button"
+                accessibilityLabel={String(month)}
+                accessibilityState={accessibilityState}
+                style={monthButtonStyle}>
+                <View style={monthInnerStyle}>
+                    {selected ? (
+                        <View style={styles.checkIconView}>
+                            <Icon name="check" size={24} />
+                        </View>
+                    ) : (
+                        <View style={styles.spacer} />
```

Comment:
I suppose this is implement for displaying the name of the month.
There is ListItem that you can use to display this dropdown.

Reply:
it doesn't have to be absolutely specific to month dropdown.
Also, add testId and name of the month to ListItem.

Reply:
done @shashank1010